### PR TITLE
Rename category mismatch to new category and update display

### LINKED ISSUE #2254: Create display for train test category mismatch
The display should be a bar chart: x-axis will be all categorical features, y-axis will be the percent of category mismatch

Hover could indicate in each bar the value of with the highest ration of category mismatch

### PR DESCRIPTION
**Pull request by matanper**
Decided to update the table instead of graph
close #2254

### CODE REVIEW

File: deepchecks/tabular/checks/train_test_validation/new_category_train_test.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+                                            'New categories ratio in data',
+                                            'New categories',
+                                            'Feature importance']).set_index(['Feature Name'])
+        result_data.sort_values(by='New categories ratio in data')
+        if all(feature_importance.isna()):
+            result_data.drop('Feature importance', axis=1, inplace=True)
+
+        if context.with_display:
+            display = result_data.copy()
+            display['New categories ratio in data'] = display['New categories ratio in data'].apply(format_percent)
```
Comment (noamzbr) [inline deepchecks/tabular/checks/train_test_validation/new_category_train_test.py:127]:
```suggestion
            display['Ratio of New Categories'] = display['Ratio of New Categories'].apply(format_percent)
```

File: deepchecks/tabular/checks/train_test_validation/new_category_train_test.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            display['# new categories'] = display['# new categories'].apply(format_number)
```
Comment (noamzbr) [inline deepchecks/tabular/checks/train_test_validation/new_category_train_test.py:128]:
```suggestion
            display['# New Categories'] = display['# New Categories'].apply(format_number)
```

File: deepchecks/tabular/checks/train_test_validation/new_category_train_test.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+            display['New categories examples'] = display['New categories']. \
```
Comment (noamzbr) [inline deepchecks/tabular/checks/train_test_validation/new_category_train_test.py:129]:
```suggestion
            display['Examples'] = display['Examples']. \
```

File: deepchecks/tabular/checks/train_test_validation/__init__.py [hunk 4]
```diff
@@ -28,6 +29,7 @@
 
 __all__ = [
     'CategoryMismatchTrainTest',
```
Comment (noamzbr) [inline deepchecks/tabular/checks/train_test_validation/__init__.py:31]: Delete this

**Review by noamzbr — COMMENTED** [review]
Add to deprecation tests

**Review by Nadav-Barak — APPROVED** [review]

**Review by noamzbr — APPROVED** [review]

**Pull request merged**